lib/hanami/model/sql/types/schema/coercions.rb

```diff
@@ -192,6 +192,12 @@ def self.hash(arg)
                 raise ArgumentError.new("invalid value for Hash(): #{arg.inspect}")
               end
             end
+
+            def self.json(arg)
```

Comment:
@katafrakt Oh right, sorry for the silly question.

I'm wondering if this is correct. `.json` can return both a `Hash` or an `Array`, according to the input. Is there the risk that `Types::Hash.constructor(Coercions.method(:json))` will receive an array?